lib/internal/readline/interface.js

```diff
@@ -769,16 +780,51 @@ class Interface extends InterfaceConstructor {
   }
 
   [kDeleteLineLeft]() {
+    const del = StringPrototypeSlice(this.line, 0, this.cursor);
     this.line = StringPrototypeSlice(this.line, this.cursor);
     this.cursor = 0;
+    this[kPushToKillRing](del);
     this[kRefreshLine]();
   }
 
   [kDeleteLineRight]() {
+    const del = StringPrototypeSlice(this.line, this.cursor);
     this.line = StringPrototypeSlice(this.line, 0, this.cursor);
+    this[kPushToKillRing](del);
     this[kRefreshLine]();
   }
 
+  [kPushToKillRing](del) {
+    if (!del || del === killRing[0]) return;
+    killRing.unshift(del);
+    while (killRing.length > kMaxLengthOfKillRing) killRing.pop();
+  }
+
+  [kYank]() {
+    if (killRing.length > 0) {
+      this[kYanking] = true;
+      this[kInsertString](killRing[0]);
+    }
+  }
+
+  [kYankPop]() {
+    if (!this[kYanking]) {
+      return;
+    }
+    if (killRing.length > 0) {
+      const lastYank = killRing[0];
+      killRing.push(killRing.shift());
```

Comment:
```suggestion
      ArrayPrototypePush(killRing, ArrayPrototypeShift(killRing));
```

And make sure both `ArrayPrototypePush` and `ArrayPrototypeShift` are included in the `primordials` imports